src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorDiagnosticsEndpoint.cs

```diff
@@ -238,49 +238,71 @@ static bool CheckIfAttributeContainsNonMarkupNodes(RazorSyntaxNode attributeNode
             }
         }
 
-        private Diagnostic[] FilterCSharpDiagnostics(Diagnostic[] unmappedDiagnostics)
+        private Diagnostic[] FilterCSharpDiagnostics(Diagnostic[] unmappedDiagnostics, RazorCodeDocument codeDocument)
         {
             return unmappedDiagnostics.Where(d =>
-                !(DiagnosticsToIgnore.Contains(d.Code?.String) &&
-                  d.Severity != DiagnosticSeverity.Error))
-                .ToArray();
+                !ShouldFilterCSharpDiagnosticBasedOnErrorCode(d, codeDocument)).ToArray();
+        }
+
+        private bool ShouldFilterCSharpDiagnosticBasedOnErrorCode(Diagnostic diagnostic, RazorCodeDocument codeDocument)
+        {
+            if (!diagnostic.Code.HasValue)
+            {
+                return false;
+            }
+
+            return diagnostic.Code.Value.String switch
+            {
+                "CS1525" => ShouldIgnoreCS1525(diagnostic, codeDocument),
+                _ => CSharpDiagnosticsToIgnore.Contains(diagnostic.Code.Value.String) &&
+                        diagnostic.Severity != DiagnosticSeverity.Error,
+            };
+
+            bool ShouldIgnoreCS1525(Diagnostic diagnostic, RazorCodeDocument codeDocument)
+            {
+                if (TryGetOriginalDiagnosticRange(diagnostic.Range, diagnostic.Severity, codeDocument, out var originalRange) &&
+                    originalRange.IsUndefined() &&
+                    CheckIfDocumentHasRazorDiagnostic(codeDocument, "RZ2008"))
```

Comment:
You should be able to reference the [diagnostic directly](https://github.com/dotnet/aspnetcore/blob/c925f99cddac0df90ed0bc4a07ecda6b054a0b02/src/Razor/Microsoft.AspNetCore.Razor.Language/src/RazorDiagnosticFactory.cs#L531) can't you?

Reply:
Put this first. Doing a range remaping isn't necessarily quick but diagnostics are pre-computed.